test/unit/path_spec.lua

```diff
@@ -293,6 +293,59 @@ describe('path_shorten_fname_if_possible', function()
   end)
 end)
 
+describe('path.c path_guess_exepath', function()
+  local cwd = lfs.currentdir()
+
+  for _,name in ipairs({'./nvim', '.nvim', 'foo/nvim'}) do
+    itp('"'..name..'" returns name catenated with CWD', function()
+      local bufsize = 255
+      local buf = cstr(bufsize, '')
+      cimp.path_guess_exepath(name, buf, bufsize)
+      eq(cwd..'/'..name, ffi.string(buf))
+    end)
+  end
+
+  itp('absolute path returns the name unmodified', function()
+    local name = '/foo/bar/baz'
+    local bufsize = 255
+    local buf = cstr(bufsize, '')
+    cimp.path_guess_exepath(name, buf, bufsize)
+    eq(name, ffi.string(buf))
+  end)
+
+  itp('returns the name unmodified if not found in $PATH', function()
+    local name = '23u0293_not_in_path'
+    local bufsize = 255
+    local buf = cstr(bufsize, '')
+    cimp.path_guess_exepath(name, buf, bufsize)
+    eq(name, ffi.string(buf))
+  end)
+
+  itp('does not crash if $PATH item exceeds MAXPATHL', function()
+    local orig_path_env = os.getenv('PATH')
+    local name = 'cat'  -- Some executable in $PATH.
+    local bufsize = 255
+    local buf = cstr(bufsize, '')
+    local insane_path = orig_path_env..':'..(("x/"):rep(4097))
+
+    cimp.os_setenv(to_cstr('PATH'), to_cstr(insane_path), true)
+    cimp.path_guess_exepath(name, buf, bufsize)
+    eq("bin/cat", string.match(ffi.string(buf), "....cat$"))
+
+    -- Restore $PATH.
```

Comment:
`itp` tests are run in a separate process, no need to bother restoring anything.